doc/developer-guides/hld/hv-startup.rst

```diff
@@ -98,40 +111,62 @@ Memory
+The Service VM or a pre-launched VM is created and launched on the physical
+CPU which configured as its vBSP. Meanwhile, for the physical CPUs which
+configured as vAPs for dedicated VMs, they will enter the default idle loop
 (refer to :ref:`VCPU_lifecycle` for details), waiting for any vCPU to be
 scheduled to them.
 
 :numref:`hvstart-vmflow` illustrates a high-level execution flow of
-creating and launching a VM, applicable to both SOS and UOS. One major
-difference in the creation of SOS and UOS is that SOS is created by the
-hypervisor, while the creation of UOSes is triggered by the DM in SOS.
+creating and launching a VM, applicable to pre-launched VM, Service VM
+and User VM. One major difference in the creation of User VM and pre-launched
+/Service VM is that pre-launched/Service VM is created by the hypervisor,
+while the creation of User VMs is triggered by the DM in Service OS.
 The main steps include:
 
 -  **Create VM**: A VM structure is allocated and initialized. A unique
-   VM ID is picked, EPT is created, I/O bitmap is set up, I/O
-   emulation handlers initialized and registered and virtual CPUID
-   entries filled. For SOS an addition e820 table is prepared.
-
--  **Create vCPUs:** Create the vCPUs, assign the physical processor it
+   VM ID is picked, EPT is initialized, e820 table for this VM is prepared,
+   I/O bitmap is set up, virtual PIC/IOAPIC/PCI/UART is initialized, EPC for
+   virtual SGX is prepared, guest PM IO is set up, IOMMU for PT dev support
+   is enabled, virtual CPUID entries are filled, and vCPUs configred in this VM's
+   ``vm config`` are prepared. For post-launched User VM, the EPT page table and
+   e820 table is actually prepared by DM instead of hypervisor.
+
+-  **Prepare vCPUs:** Create the vCPUs, assign the physical processor it
    is pinned to, a unique-per-VM vCPU ID and a globally unique VPID,
-   and initializes its virtual lapic and MTRR. For SOS one vCPU is
-   created for each physical CPU on the platform. For UOS the DM
-   determines the number of vCPUs to be created.
-
--  **SW Load:** The BSP of a VM also prepares for each VM's SW
-   configuration including kernel entry address, ramdisk address,
-   bootargs, zero page etc. This is done by the hypervisor for SOS
-   while by DM for UOS.
+   and initializes its virtual lapic and MTRR, and its vCPU thread object got setup
+   for vcpu scheduling. The vCPU number and affinity are defined in corresponding
+   ``vm config`` for this VM.
+
+-  **Build vACPI:** For Service VM, the hypervisor will customize a virtual ACPI
+   table based on native ACPI table (this is in the TODO).
+   For pre-launched VM, the hypervisor will build a simple ACPI table with necessary
+   information like MADT.
+   For post-launched User VM, DM will build its ACPI table dynamically.
+
+-  **SW Load:** Prepares for each VM's SW configuration including kernel entry
+   address, ramdisk address, bootargs, or zero page for launching bzImage etc.
```

Comment:
What about change this to:
Prepare for each VM's SW configuration according to guest OS requirement?

The reason is that we are not only for linux kernel but also for pre-launched OS which may have different configuration info.